functional/pipeline.py

```diff
@@ -1028,20 +1012,10 @@ def product(self, projection=None):
         :return: product of elements in sequence
         """
         if self.empty():
-            if projection:
-                return projection(1)
-            else:
-                return 1
+            return projection(1) if projection else 1
```

Comment:
Still prefer prior style here, but as before could see accepting if there is advantage to it

Reply:
there is no real advantage (maybe very slightly easier for static analysis tools like mypy? not sure), I will revert it